build_path.go

```diff
@@ -82,15 +82,19 @@ func buildOperation(ws *restful.WebService, r restful.Route) *spec.Operation {
 	return o
 }
 
-func buildParameter(r *restful.Parameter) spec.Parameter {
+func buildParameter(r restful.Route, restfulParam *restful.Parameter) spec.Parameter {
 	p := spec.Parameter{}
-	param := r.Data()
+	param := restfulParam.Data()
 	p.In = asParamType(param.Kind)
 	p.Description = param.Description
 	p.Name = param.Name
 	p.Required = param.Required
 	p.Default = param.DefaultValue
 	p.Format = param.DataFormat
+	if p.In == "body" && r.ReadSample != nil && p.Type == reflect.TypeOf(r.ReadSample).String() {
```

Comment:
I'm not actually sure if this is allowed by the schema, but I wanted to make sure that if `p.Type` was a primitive, like `string`, we didn't set `p.Schema.Ref` to `#/definitions/string`.